src/core/lombok/javac/handlers/HandleSuperBuilder.java

```diff
@@ -1160,7 +1160,7 @@ private boolean constructorExists(JavacNode type, String builderClassName) {
 						if (lastIndexOfDot >= 0) {
 							typeName = typeName.substring(lastIndexOfDot+1);
 						}
-						if ((builderClassName+"<?, ?>").equals(typeName))
+						if (typeName.startsWith(builderClassName) && typeName.endsWith("?, ?>"))
```

Comment:
The code before was not optimal already, as String comparisons tend to break quickly. Imagine something like Java extending the diamond operator also for ?, so that you can write:
`protected Parent(ParentBuilder<> b) {...}`
Thus, we should use the opportunity to improve it. I suggest doing an `instanceof` check and only use the generic `toString` as a fallback:
```
String typeName;
JCTree paramType = md.params.get(0).getType();
if (paramType instanceof JCTypeApply) {
	typeName = ((JCTypeApply)paramType).clazz.toString();
} else {
	typeName = paramType.toString();
}
int lastIndexOfDot = typeName.lastIndexOf('.');
if (lastIndexOfDot >= 0) {
	typeName = typeName.substring(lastIndexOfDot+1);
}
if (builderClassName.equals(typeName))
	return true;
```

Reply:
Thanks, I'll work on that. I won't like have a chance until a couple weeks from now.

Reply:
Yeah, using `toString()` to compare generics is a nono, though I find the idea of falling back on that in case a series of `instanceof` fails a great idea. @ntroutman still working on it, or?